youtube_dl/extractor/soundcloud.py

```diff
@@ -402,19 +402,17 @@ def resolve_permalink_url(candidates):
                             return permalink_url, entry_id, title
 
             for e in collection:
-                tr_dict = e.get('track')
-                pl_dict = e.get('playlist')
+                extractor = None
+                if isinstance(e.get('track'), dict):
```

Comment:
This does not guarantee you anything. What you should do is to check `suitable` on resolved URL.